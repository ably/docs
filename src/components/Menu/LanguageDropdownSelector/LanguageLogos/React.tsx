import React from 'react';

// The best one.
export const ReactJS = () => (
  <svg width="24" height="24" viewBox="-11.5 -10.23174 23 20.46348" fill="none" xmlns="http://www.w3.org/2000/svg">
    <circle cx="0" cy="0" r="2.05" fill="#61dafb" />
    <g stroke="#61dafb" strokeWidth="1" fill="none">
      <ellipse rx="11" ry="4.2" />
      <ellipse rx="11" ry="4.2" transform="rotate(60)" />
      <ellipse rx="11" ry="4.2" transform="rotate(120)" />
    </g>
  </svg>
);
