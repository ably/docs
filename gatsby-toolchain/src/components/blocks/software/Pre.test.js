import React from 'react';
import { Pre } from '.';
import TestRenderer from 'react-test-renderer';

const preRootElement = (
  <Pre
    data={[{ data: 'var useVars = false;', type: 'text' }]}
    language="javascript"
    languages={['javascript', 'csharp']}
    altData={{
      javascript: [{ data: 'var useVars = false;', type: 'text' }],
      csharp: [{ data: 'bool useVars = false;', type: 'text' }],
    }}
  />
);

describe('Different langs provided to Code elements results in different rendering', () => {
  const preRootElementRenderer = TestRenderer.create(preRootElement);
  const preJSON = preRootElementRenderer.toJSON();
  it('Successfully renders Code elements with language', () => {
    expect(preJSON).toMatchSnapshot();
  });
});
