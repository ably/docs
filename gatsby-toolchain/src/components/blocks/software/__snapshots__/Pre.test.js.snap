// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Different langs provided to Code elements results in different rendering Successfully renders Code elements with language 1`] = `
<pre
  className="mt-32 mb-0"
>
  <menu
    className="sc-egiyK sc-bqiRlB sc-ksdxgE iXgqax fejsjT jHWssQ"
  >
    <li
      className="c-menu__item c-menu__label"
    >
      No JS example exists, select:
    </li>
    <li
      className="c-menu__item"
    >
      <button
        className="sc-kDTinF sc-iqseJM sc-crHmcD bIVlGb fNSJkt gEXRWt"
        onClick={[Function]}
        value="javascript"
      >
        JS
      </button>
    </li>
    <li
      className="c-menu__item"
    >
      <button
        className="sc-kDTinF sc-iqseJM bIVlGb fNSJkt"
        onClick={[Function]}
        value="csharp"
      >
        C#
      </button>
    </li>
  </menu>
  var useVars = false;
</pre>
`;
